sdk/core/core-http/src/credentials/credentials.ts

```diff
@@ -1,4 +1,7 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+/**
+ * A function that receives a challenge and resolves a promise with a string token.
```

Comment:
When does the Authenticator gets used? Refer to that instead of repeating what the code says.